### ISSUE TITLE
Support colsample_bylevel as in xgboost

### ISSUE THREAD

**Issue by gugatr0n1c** [Microsoft/LightGBM#98]
Hi,
there is nice parameter: colsample_bylevel

notes here:
https://github.com/dmlc/xgboost/blob/master/doc/parameter.md

Which can probably speed training up even more. And I have tested that it help deal with overfitting as well, and on my datasets it can improve accuracy.

Can you support this here as well?

thx

**Comment by guolinke** [comment]
@gugatr0n1c  you can try ```feature_fraction''' parameter.
It is column sample by tree. 
And LightGBM doesn't have level concept.

**Comment by gugatr0n1c** [comment]
yeah, I am using feature_fraction. But when I am using hyperparameters tuning in xgb, there is always significant to  use all three: bagging rows, subsample on columns and this one.

I understand that LightGBM does not use levels, so  maybe this can be done in each split? 
This is also good feature for randomforest mode, where you can achieve with this even more random forest.
But if this can slow down training, it is of course trade off.

**Comment by guolinke** [comment]
@gugatr0n1c 
LightGBM use Histogram subtraction to speed up training. 
It will loss on speed if we use features randomly at every split, for we cannot cache histogram for some features.

However, I will have a try.

**Comment by gugatr0n1c** [comment]
ok I understand... thx for clearing this up for me... thx for trying, however this is not "critical" feature to have, so feel free to throw away this idea..

**Comment by guolinke** [comment]
close for now. It is not efficient to do this.
